src/main/java/net/rptools/maptool/client/functions/LookupTableFunction.java

```diff
@@ -84,367 +90,521 @@ public Object childEvaluate(
+   *
+   * @param function name of this MTscript function, "{@code loadTable}"
+   * @param params parameters of the MTscript function
+   * @return a number indicating the successful record count, or a {@link JsonArray} of all elements
+   *     that failed to load
+   * @throws ParserException Thrown when syntax errors are detected in the input data. This includes
+   *     negative numbers in either of the numeric fields of a table entry, a lower die roll that is
+   *     greater than the upper die roll of a table entry and vice versa, missing fields, too many
+   *     fields, and potentially many more.
+   */
+  private @NotNull JsonElement loadTable(String function, List<Object> params)
+      throws ParserException {
+    checkTrusted(function);
+    FunctionUtil.checkNumberParam(function, params, 3, 3);
+    String name = params.get(0).toString();
+    LookupTable lookupTable = checkTableAccess(name, function);
+
+    Object data = params.get(1);
+    if (data instanceof String) {
+      data = com.google.gson.JsonParser.parseString(data.toString());
+    }
+    if (data instanceof JsonArray json) {
+      JsonArray errorRows = new JsonArray();
+      long counter = 0, rowindex = 0;
+      // Verify that all records are themselves JSONArray objects and each has 4 fields
+      for (JsonElement row : json) {
         try {
-          BigDecimal bival = new BigDecimal(val);
-          return bival;
-        } catch (NumberFormatException nfe) {
-          return val;
+          if (row instanceof JsonArray array && array.size() == 4) {
+            int min = array.get(0).getAsInt();
+            int max = array.get(1).getAsInt();
+            String value = array.get(2).getAsString();
+            String image = array.get(3).getAsString();
+            if (min >= 1 && max >= min && !value.isEmpty()) {
+              // Image is allowed to be empty, but if given, it must be an asset ID
+              MD5Key imageID = null;
+              if (!image.isEmpty()) {
+                imageID = FunctionUtil.getAssetKeyFromString(image);
+              }
+              // All checks complete, add this row to the table
+              lookupTable.addEntry(min, max, value, imageID);
+              counter++;
+            }
+          }
+        } catch (UnsupportedOperationException | NumberFormatException | IllegalStateException e) {
+          // Make a list of all failing rows to return to the user.
+          errorRows.add(rowindex);
         }
-      } else if ("tableImage".equalsIgnoreCase(function)
-          || "tblImage"
-              .equalsIgnoreCase(function)) { // We want the image URI through tblImage or tableImage
+        rowindex++;
+      }
+      MapTool.serverCommand().updateCampaign(MapTool.getCampaign().getCampaignProperties());
+      return counter == rowindex ? new JsonPrimitive(counter) : errorRows;
+    }
+    // FIXME Report error using I18N
+    throw new ParserException("Second parameter must be a JSON Array");
```

Comment:
Should be extracted to i18n

Reply:
@Azhrei I think this may work (so as to maybe get this in to 1.15.1.
`throw new ParserException(
                I18N.getText("macro.function.json.onlyArray"));`